fs/rc/rcserver/rcserver_test.go

```diff
@@ -847,3 +848,76 @@ func TestServeModTime(t *testing.T) {
 	}}
 	testServer(t, tests, &opt)
 }
+
+func TestContentTypeJSON(t *testing.T) {
+	tests := []testRun{
+		{
+			Name:        "Check Content-Type for JSON response",
+			URL:         "rc/noop",
+			Method:      "POST",
+			Body:        `{}`,
+			ContentType: "application/json",
+			Status:      http.StatusOK,
+			Expected:    "{}\n",
+			Headers: map[string]string{
+				"Content-Type": "application/json",
+			},
+		},
+	}
+	opt := newTestOpt()
+	testServer(t, tests, &opt)
+}
+
+func TestErrorContentTypeJSON(t *testing.T) {
+	tests := []struct {
+		Name           string
+		URL            string
+		Method         string
+		Body           string
+		ExpectedStatus int
+		ExpectedJSON   string
+	}{
+		{
+			Name:           "Check Content-Type for JSON error response",
+			URL:            "rc/error",
+			Method:         "POST",
+			Body:           `{}`,
+			ExpectedStatus: http.StatusInternalServerError,
+			ExpectedJSON: `{
+				"error": "arbitrary error on input map[]",
+				"input": {},
+				"path": "rc/error",
+				"status": 500
+			}`,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.Name, func(t *testing.T) {
+			req, err := http.NewRequest(tc.Method, tc.URL, nil)
+			require.NoError(t, err)
+
+			rr := httptest.NewRecorder()
+			handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				// Simulate setting the error response in your actual handler
+				w.Header().Set("Content-Type", "application/json")
+				w.WriteHeader(tc.ExpectedStatus)
+				w.Write([]byte(tc.ExpectedJSON)) // This should be replaced with the actual error handling logic
```

Comment:
The linter says this is missing an error check. `w.Write` returns a length and error if I remember correctly, so you'd check it something like this

```go
_, err := w.Write(...)
require.NoError(t, err)
```